mesonbuild/compilers/compilers.py

```diff
@@ -1141,19 +1141,35 @@ def has_func_attribute(self, name, env):
         raise EnvironmentException(
             'Language {} does not support function attributes.'.format(self.get_display_language()))
 
-GCC_STANDARD = 0
-GCC_OSX = 1
-GCC_MINGW = 2
-GCC_CYGWIN = 3
 
-CLANG_STANDARD = 0
-CLANG_OSX = 1
-CLANG_WIN = 2
-# Possibly clang-cl?
+@enum.unique
+class CompilerType(enum.Enum):
+    GCC_STANDARD = 0
+    GCC_OSX = 1
+    GCC_MINGW = 2
+    GCC_CYGWIN = 3
+
+    CLANG_STANDARD = 10
+    CLANG_OSX = 11
+    CLANG_WIN = 12
```

Comment:
This sounds like a good time to rename this to `CLANG_MINGW` because that's what it actually is.

Reply:
Changed to `CLANG_MINGW`